### PR TITLE
feat: add vitest matchers

### PR DESCRIPTION
**Pull request by luxass**
### 📚 Description

This PR adds custom Vitest Matchers. I couldn't get the tsconfig's in each package to pickup the types file.

I have tried by setting the types in the base.json, that didn't work. typeRoots doesn't work either. The only thing i have found that make it works, is by including the types file in the includes list or setting a reference at the top of the files using it.

By in the future, we may wanna move test-utils out of the monorepo. There is a lot of edge cases around having it inside the monorepo. Such as we are always using the direct source, and not the actual api exposed.

But before we can do that, we will need to merge `ucd-store-v2` into `ucd-store`, and release all the packages.

### CODE REVIEW

File: packages/fs-bridge/test/security/node/unc-paths.test.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/// <reference types="../../../../test-utils/src/matchers/types.d.ts" />
```
[the remark this answers is not in the document]
Comment (luxass) [reply packages/fs-bridge/test/security/node/unc-paths.test.ts]: I have tried typeRoots, that didn't work! Since the global setup file will not be picked up by TypeScripts, it will not work to have it there.
